### PR TITLE
#811, add get_full_accounts api in cli_wallet

### PR DESCRIPTION
**Pull request by cifer-lee**

### CODE REVIEW

File: libraries/wallet/wallet.cpp [hunk 1]
```diff
@@ -2924,6 +2924,11 @@ account_history_operation_detail wallet_api::get_account_history_by_operations(s
     return result;
 }
 
+std::map<string,full_account> wallet_api::get_full_accounts( const vector<string>& names_or_ids, bool subscribe)
+{
+    return my->_remote_db->get_full_accounts(names_or_ids, false);
```
Comment (oxarbitrage) [inline libraries/wallet/wallet.cpp]: here the subscribe bool will be always off.

Comment (cifer-lee) [reply libraries/wallet/wallet.cpp]: ah oh.. how careless I am, thanks for pointing out~ have fixed and pushed again

**Comment by abitmore** [comment]
Actually I don't know who is going to use this API. If used as a command, the output is not formatted, so hard to read; if used with RPC, why not query witness_node directly?

File: libraries/wallet/include/graphene/wallet/wallet.hpp [hunk 2]
```diff
@@ -377,6 +377,7 @@ class wallet_api
        */
      vector<operation_detail>  get_relative_account_history(string name, uint32_t stop, int limit, uint32_t start)const;
 
+      std::map<string,full_account>     get_full_accounts( const vector<string>& names_or_ids, bool subscribe);
```
Comment (abitmore) [inline libraries/wallet/include/graphene/wallet/wallet.hpp]: Need doc.

Comment (cifer-lee) [reply libraries/wallet/include/graphene/wallet/wallet.hpp]: have added the doc

**Comment by cifer-lee** [comment]
@abitmore thanks for pointing out, I have added the doc and remove the unneeded `subscribe` parameter

File: libraries/wallet/include/graphene/wallet/wallet.hpp [hunk 3]
```diff
@@ -377,7 +377,17 @@ class wallet_api
        */
      vector<operation_detail>  get_relative_account_history(string name, uint32_t stop, int limit, uint32_t start)const;
 
-      std::map<string,full_account>     get_full_accounts( const vector<string>& names_or_ids, bool subscribe);
+      /**
+       * @brief Fetch all objects relevant to the specified accounts and subscribe to updates
+       * @param names_or_ids Each item must be the name or ID of an account to retrieve
+       * @return Map of string from @ref names_or_ids to the corresponding account
+       *
+       * This function fetches all relevant objects for the given accounts, and subscribes to updates to the given
```
Comment (abitmore) [inline libraries/wallet/include/graphene/wallet/wallet.hpp]: Still `subscribes`?

File: libraries/wallet/include/graphene/wallet/wallet.hpp [hunk 4]
```diff
@@ -377,7 +377,17 @@ class wallet_api
...
+       * accounts. If any of the strings in @ref names_or_ids cannot be tied to an account, that input will be
+       * ignored. All other accounts will be retrieved and subscribed.
```
Comment (abitmore) [inline libraries/wallet/include/graphene/wallet/wallet.hpp]: Also remove mention of "subscribed" here.

File: libraries/wallet/include/graphene/wallet/wallet.hpp [hunk 5]
```diff
@@ -377,7 +377,17 @@ class wallet_api
        */
      vector<operation_detail>  get_relative_account_history(string name, uint32_t stop, int limit, uint32_t start)const;
 
-      std::map<string,full_account>     get_full_accounts( const vector<string>& names_or_ids, bool subscribe);
+      /**
+       * @brief Fetch all objects relevant to the specified accounts and subscribe to updates
```
Comment (abitmore) [inline libraries/wallet/include/graphene/wallet/wallet.hpp]: Remove "subscribe".

Comment (cifer-lee) [reply libraries/wallet/include/graphene/wallet/wallet.hpp]: sorry, it has been removed now

File: libraries/wallet/include/graphene/wallet/wallet.hpp [hunk 6]
```diff
@@ -378,13 +378,12 @@ class wallet_api
      vector<operation_detail>  get_relative_account_history(string name, uint32_t stop, int limit, uint32_t start)const;
 
       /**
-       * @brief Fetch all objects relevant to the specified accounts and subscribe to updates
+       * @brief Fetch all objects relevant to the specified accounts to updates
```
Comment (abitmore) [inline libraries/wallet/include/graphene/wallet/wallet.hpp]: Please remove "to updates" as well

**Comment by cifer-lee** [comment]
I noticed someone requested this api in https://github.com/bitshares/bitshares-core/issues/811, so added it...

**Comment by oxarbitrage** [comment]
the usage i find for this is that you(person) are in the wallet and need to get everything from an account. with this you can make it without leaving the wallet. i agree the value is limited and pretty much none for programs.

on here: https://github.com/bitshares/bitshares-core/issues/811#issuecomment-378572556 i suggest a more generic api to get anything from the database api but this comment didn't got any feedback. that should increase usability.

to remain compatible there is 1 thing we should consider in this case:

i am comparing:

` account_object get_account(string account_name_or_id)`

with:

`std::map<string, full_account> get_full_accounts(const vector<string> & names_or_ids)`

no need for the argument to be an array in the wallet IMO. the output for pulling 1 account at a time is already big enough to return more than one `full_account` in the same call. 

so it can be:

`full_account get_full_accounts(string name_or_id)`

and then cast to array of 1 element when calling the database api.

this pull request as it is now will close the issue and i have no problems on merging but some discussion about those pointed issues will make no harm.

**Comment by cifer-lee** [comment]
@oxarbitrage agree with you, I will update to return `full_account` instead of `std::map<string, full_account>` later

**Pull request merged**